base_layer/core/src/base_node/tari_pulse_service/mod.rs

```diff
@@ -0,0 +1,190 @@
+use std::{str::FromStr, time::Duration};
+
+use futures::future;
+use hickory_client::{
+    client::{AsyncDnssecClient, ClientHandle},
+    proto::{
+        iocompat::AsyncIoTokioAsStd,
+        rr::dnssec::{public_key::Rsa, SigSigner, TrustAnchor},
+        xfer::DnsMultiplexer,
+    },
+    rr::{DNSClass, Name, RData, Record, RecordType},
+    tcp::TcpClientStream,
+};
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+use tari_service_framework::{async_trait, ServiceInitializationError, ServiceInitializer, ServiceInitializerContext};
+use tari_shutdown::ShutdownSignal;
+use tari_utilities::hex::Hex;
+use tokio::{net::TcpStream as TokioTcpStream, sync::watch, time};
+
+use super::LocalNodeCommsInterface;
+use crate::base_node::comms_interface::CommsInterfaceError;
+
+const LOG_TARGET: &str = "c::bn::tari_pulse";
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TariPulseConfig {
+    pub check_interval: Option<Duration>,
```

Comment:
```suggestion
    pub check_interval:Duration,
```
then you can get rid of the unwrap
And implement default, so you can just call default to construct it

Reply:
Removed unnecesary wrapper